x/credit/internal/keeper/keeper.go

```diff
@@ -0,0 +1,97 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/mesg-foundation/engine/x/credit/internal/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// Keeper of the credit store
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      *codec.Codec
+}
+
+// NewKeeper creates a credit keeper
+func NewKeeper(cdc *codec.Codec, key sdk.StoreKey) Keeper {
+	keeper := Keeper{
+		storeKey: key,
+		cdc:      cdc,
+	}
+	return keeper
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+// Add a number of credits to an address
+func (k Keeper) Add(ctx sdk.Context, address sdk.AccAddress, amount sdk.Int) (sdk.Int, error) {
+	value, err := k.Get(ctx, address)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	res := value.Add(amount)
+	return k.Set(ctx, address, res)
+}
+
+// Sub a number of credits to an address
+func (k Keeper) Sub(ctx sdk.Context, address sdk.AccAddress, amount sdk.Int) (sdk.Int, error) {
+	value, err := k.Get(ctx, address)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	res := value.Sub(amount)
+	return k.Set(ctx, address, res)
+}
+
+// Transfer credits from one address to another
+func (k Keeper) Transfer(ctx sdk.Context, from, to sdk.AccAddress, amount sdk.Int) error {
+	if _, err := k.Sub(ctx, from, amount); err != nil {
+		return err
+	}
+	if _, err := k.Add(ctx, to, amount); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Set a number of credit to a specific address
+func (k Keeper) Set(ctx sdk.Context, address sdk.AccAddress, amount sdk.Int) (sdk.Int, error) {
```

Comment:
Let's rename amount to balance to be more explicit about what is it.
```suggestion
func (k Keeper) Set(ctx sdk.Context, address sdk.AccAddress, balance sdk.Int) (sdk.Int, error) {
```